### ISSUE TITLE
WAN 2.1 LoRA Training Time Sanity Check

### ISSUE THREAD

**Issue by samuelm2** [tdrussell/diffusion-pipe#152]
Hi! I'm currently training my first WAN 2.1 LoRA using diffusion-pipe. It is such a useful tool, thank you so much for building it!

I had some questions about expected training times, as I wanted to make sure that the times that I'm seeing are normal or if I have something wrong with my setup. Right now, I'm training using an A6000 (48GB VRAM) on Runpod with 16 vCPUs and 62GB RAM.

I'm training using Wan2.1-I2V-14B-480P (taken from HuggingFace here: https://huggingface.co/Wan-AI/Wan2.1-I2V-14B-480P ). My input is 10 videos, each with 63 frames at 15FPS. I'm using mostly the defaults found in the example [hunyuan_video.toml](https://github.com/tdrussell/diffusion-pipe/blob/main/examples/hunyuan_video.toml), except with the [model] section switched with the example [wan2.1 toml](https://github.com/tdrussell/diffusion-pipe/blob/main/docs/supported_models.md#wan21)

Additionally, I made the following changes:
- Increased the learning rate from `2e-5` to `6e-5`
- In dataset.toml, swapped out the frame_buckets from `[1, 33]` to `[1, 33, 63]` because I want the lora to work best at 63 frames.
- Set `blocks_to_swap = 2` because I was going OOM after I modified `frame_buckets`

With these settings and configuration, I'm getting around 120 to 180 seconds per step, so it takes several hours to train 10 epochs. The GPU usage and memory are both max'ed out whenever I run `nvidia-smi`. 

The slowdown really seemed to come whenever I switched the frame_buckets from [1, 33] to [1,33,63], that's when I started running OOM and had to set blocks_to_swap. Before that, I was getting around 60-90 seconds per step.

Is this what others are seeing? What training times / specs is everyone else seeing for WAN 2.1? 

In a comment on CivitAI, someone mentioned they were able to train a WAN 2.1 720p LoRA with a 4090 for 32 epochs in 35 minutes using Musubi-Tuner with default parameters which seems crazy so I'm not sure if it is accurate or if I'm way off somehow.

Thank you for the help, and let me know if you have any questions or suggestions.

**Comment by Bobothesecond** [comment]
You and I may have read the same article because I also saw the same crazy numbers. For the record I am on two 3090s and with a bucket size of 1, 17 and 33 I get around right on 60 seconds per step. So either we're doing something wrong or something is up.

**Comment by samuelm2** [comment]
I ended up downscaling the input videos to 256x256, which allowed me to run without using blocks_to_swap. Now its much faster, at around ~25 to 35 seconds per step. We'll see how the lora quality turns out

**Comment by Bobothesecond** [comment]
Interesting, do you still train at 512 resolution or did you change that too?

**Comment by samuelm2** [comment]
I changed that too -- so now I'm training at 256 resolution

**Comment by tdrussell** [comment]
On a 4090 power limited to 350W, using blocks_to_swap=32 I can train 512x512x63 video at 24 seconds per step, using a global batch size of 1. Even with the block swapping it is almost maxing out the 350W limit.

If you changed nothing in the example config, it has gradient_accumulation_steps=4, so your steps would take 4 times as long (of course it is now seeing 4 examples in a step instead of 1).

4090 should be a little less than double the compute of a A6000. So doing 24 * 4 * 1.8 = 172 seconds per step, this rough estimation is in line with the numbers you report.

> WAN 2.1 720p LoRA with a 4090 for 32 epochs in 35 minutes

This could be possible if the dataset if very small (like 10 videos), and also the training resolution and length are much smaller. Remember that attention is quadratic time complexity.

Also did you leave num_repeats=10? If so, 1 epoch is actually 10 epochs worth of data. Maybe I should change the example file to have it at 1, but the problem is many users use very small datasets, and without repeats, a significant fraction of the dataset could be dropped due to the aspect ratio bucketing if the global batch size is >1.

**Comment by alienzed** [comment]
tdrussell said it all, but yeah... 512x512x63 is huge my friends... Those times are normal imo.
384x384x50 takes me over 30 seconds per step when training hunyuanvideo.
